### PR TITLE
ref(apidocs-testing): Releases Endpoint Testing

### PR DESCRIPTION
**Pull request by MeredithAnya**
**Context:**
Now that the schema for the documented API endpoints is in Sentry, we have been writing tests to make sure that if an endpoint is changed, we can catch if the change would violate the schema we have in place. This makes sure that there are not accidental breaking changes to these document endpoints that would affect the developers who build on top of them. 

**Release Endpoints**
The latest and greatest in the batches of tests, this PR covers the tests for the endpoints in the "Releases" section of the API docs. 

Some things to take note of:
* The tests found that some status codes were wrong (`201` instead of `200`).
* Reference objects (`$ref`) do not seem to work with `"nullable": true`, there is a [whole discussion](https://github.com/OAI/OpenAPI-Specification/issues/1368) about it but even with some of the suggestions I couldn't get the openapi-core validator to work correctly. Further research required, but for now I've added the whole `Deploy` object instead of referencing it.
* Not adding `delete` paths since we don't get a response payload back.

### CODE REVIEW

File: tests/apidocs/endpoints/releases/test_organization_release_details.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from sentry.models import Release
+from tests.apidocs.util import APIDocsTestCase
+
+
+class OrganizationReleaseDetailsDocsTest(APIDocsTestCase):
+    def setUp(self):
+        user = self.create_user(is_staff=False, is_superuser=False)
+        org = self.organization
+        org2 = self.create_organization()
+        org.flags.allow_joinleave = False
```
Comment (ceorourke) [inline tests/apidocs/endpoints/releases/test_organization_release_details.py:17]: what is this part for?

File: tests/apidocs/endpoints/releases/test_organization_release_details.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            "sentry-api-0-organization-release-details",
+            kwargs={"organization_slug": org.slug, "version": release1.version},
+        )
+
+    def test_get(self):
+        response = self.client.get(self.url)
+        request = RequestFactory().get(self.url)
+
+        self.validate_schema(request, response)
+        pass
```
Comment (ceorourke) [inline tests/apidocs/endpoints/releases/test_organization_release_details.py:51]: why is there a `pass` here?

File: tests/apidocs/endpoints/releases/test_organization_release_file_details.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        release.add_project(project)
+
+        file1 = self.create_file(
+            name="blah.js",
+            size=42,
+            type="release.file",
+            headers={"Content-Type": "application/json"},
+            checksum="dc1e3f3e411979d336c3057cce64294f3420f93a",
+        )
+        releasefile = ReleaseFile.objects.create(
```
Comment (ceorourke) [inline tests/apidocs/endpoints/releases/test_organization_release_file_details.py:26]: you can use the `create_release_file` fixture here too

File: tests/apidocs/endpoints/releases/test_organization_releases.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        self.url = reverse(
+            "sentry-api-0-organization-releases", kwargs={"organization_slug": org.slug}
+        )
+
+    def test_get(self):
+        response = self.client.get(self.url)
+        request = RequestFactory().get(self.url)
+
+        self.validate_schema(request, response)
+        pass
```
Comment (ceorourke) [inline tests/apidocs/endpoints/releases/test_organization_releases.py:50]: there's a `pass` here too

File: tests/apidocs/endpoints/releases/test_project_issues_solved_in_release.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        self.user = self.create_user()
+        self.org = self.create_organization()
+        self.team = self.create_team(organization=self.org)
+        self.create_member(organization=self.org, user=self.user, teams=[self.team])
+        self.project = self.create_project(teams=[self.team])
+        self.release = self.create_release(project=self.project)
+        self.group = self.create_group(project=self.project)
+        self.login_as(self.user)
+
+        repo = Repository.objects.create(organization_id=self.org.id, name=self.project.name)
```
Comment (ceorourke) [inline tests/apidocs/endpoints/releases/test_project_issues_solved_in_release.py:27]: can use `create_repo` here

**Review by ceorourke — CHANGES_REQUESTED** [review]
Overall looks good, the fixtures are optional but I think the 2 `pass` statements left in need to be removed.

File: tests/apidocs/endpoints/releases/test_organization_release_details.py (hunk 2 above)
Comment (MeredithAnya) [reply tests/apidocs/endpoints/releases/test_organization_release_details.py]: leftover oopsie :)

**Comment by MeredithAnya** [comment]
> Overall looks good, the fixtures are optional but I think the 2 `pass` statements left in need to be removed.

The fixtures is a good call cause if tables change I'm going assume the fixtures will be updated.

**Review by ceorourke — APPROVED** [review]
👍

**Comment by MeredithAnya** [comment]
@ceorourke I'm adding this comment more as taking note, but I didn't use the `create_commit` factory because I ran into some problems with it auto creating the commit release files. Probs doing something wrong but might follow up later with those changes

**Pull request merged**